### PR TITLE
fix default language fallback

### PR DESCRIPTION
**Pull request by 2403905**
## Description

Added the global variable OCIS_DEFAULT_LANGUAGE. Now `settings`, `notification`, and `userlog` services support the default language fallback.

The `OCIS_DEFAULT_LANGUAGE` impacts depend on existing translation.

These changes are needed because if the transaction does not exist for notification or userlog services we want to fall down to the OCIS_DEFAULT_LANGUAGE first and after that to the English

## Related Issue

- Fixes [https://github.com/owncloud/ocis/issues/7465](https://github.com/owncloud/ocis/issues/7465)

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/userlog/pkg/config/config.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ type Config struct {
 
 	RevaGateway     string      `yaml:"reva_gateway" env:"OCIS_REVA_GATEWAY" desc:"CS3 gateway used to look up user metadata"`
 	TranslationPath string      `yaml:"translation_path" env:"OCIS_TRANSLATION_PATH;USERLOG_TRANSLATION_PATH" desc:"(optional) Set this to a path with custom translations to overwrite the builtin translations. Note that file and folder naming rules apply, see the documentation for more details."`
+	DefaultLanguage string      `yaml:"default_language" env:"OCIS_DEFAULT_LANGUAGE,USERLOG_DEFAULT_LANGUAGE" desc:"(optional) The default language. If not defined, English will be used as default. See the documentation for more details."`
```
Comment (mmattel) [inline services/userlog/pkg/config/config.go:27]:
```suggestion
	DefaultLanguage string      `yaml:"default_language" env:"OCIS_DEFAULT_LANGUAGE,USERLOG_DEFAULT_LANGUAGE" desc:"(optional) The default language. If not defined, English will be used as default. See the 'settings' service documentation for more details."`
```

File: services/notifications/pkg/config/config.go [hunk 2]
```diff
@@ -31,6 +32,7 @@ type Notifications struct {
 	Events            Events                `yaml:"events"`
 	EmailTemplatePath string                `yaml:"email_template_path" env:"OCIS_EMAIL_TEMPLATE_PATH;NOTIFICATIONS_EMAIL_TEMPLATE_PATH" desc:"Path to Email notification templates overriding embedded ones."`
 	TranslationPath   string                `yaml:"translation_path" env:"OCIS_TRANSLATION_PATH,NOTIFICATIONS_TRANSLATION_PATH" desc:"(optional) Set this to a path with custom translations to overwrite the builtin translations. Note that file and folder naming rules apply, see the documentation for more details."`
+	DefaultLanguage   string                `yaml:"default_language" env:"OCIS_DEFAULT_LANGUAGE,NOTIFICATIONS_DEFAULT_LANGUAGE" desc:"(optional) The default language. If not defined, English will be used as default. See the documentation for more details."`
```
Comment (mmattel) [inline services/notifications/pkg/config/config.go:35]:
```suggestion
	DefaultLanguage   string                `yaml:"default_language" env:"OCIS_DEFAULT_LANGUAGE,NOTIFICATIONS_DEFAULT_LANGUAGE" desc:"(optional) The default language. If not defined, English will be used as default. See the 'settings' service documentation for more details."`
```
There is no docs in the notifications readme, but in the settings service readme...

**Comment by mmattel** [comment]
I am not convinced about:
* `(optional) The default language.` text because it does not tell briefly about for what it is gonna used for. Also `(optional)` is not good.
* referencing to the settings service documentation for more details of the envvar BUT without linking which is just in this case not possible (dev-docs != admin docs). Maybe we should duplicate the referenced text to the readme's accordingly as we have in othe rservices too (see auth-service, userlog, clientlog).

Thinking about how we can improve...

File: changelog/unreleased/fix-default-mail-language-fallback.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Bugfix: Fix default language fallback
+
+Add the default language for the userlog and notification
```
Comment (mmattel) [inline changelog/unreleased/fix-default-mail-language-fallback.md:3]:
```suggestion
Add the default language for the settings, userlog and notification service
```

**Comment by mmattel** [comment]
I will make a local branch, do the doc changes there and commit them. This makes it easier for me testing the rendering.

**Comment by mmattel** [comment]
@2403905 I have rebased and updated the texts via commits above, pls rebase locally.

**Review by kobergj — APPROVED** [review]

**Pull request merged**